docs/docsite/rst/dev_guide/developing_program_flow_modules.rst

```diff
@@ -766,6 +766,23 @@ This section will discuss the behavioral attributes for arguments:
 
   If ``options`` is specified, ``required_by`` refers to the sub-options described in ``options`` and behaves as in :ref:`argument_spec_dependencies`.
 
+:context:
+
+  .. versionadded:: 2.17
+
+  This key accepts an arbitrary dict of additional context utilized by the module for a specific option.
```

Comment:
I feel like this should be expanded a bit to make it clear "[that 'context' allows for custom content that will be ignored by the core engine](https://github.com/ansible/ansible/pull/82183#issuecomment-1806043529)"